### ISSUE TITLE
Linux support?

### ISSUE THREAD

**Issue by neuromancer** [LRFLEW/OpenMC2#1]
Hi,

Is Linux support planned?

Thanks!

**Comment by janisozaur** [comment]
I'm quite sure it will come, but at this early stage there's a lot of ground work to be done. To list a few hurdles:
- MS ABI, which is quite different from what Linux uses
  - This includes exception handling
- DirectX
- All the other code reaching out to system.

It will take some time. For probably a smaller project, OpenRCT2, it took around a year to get working on Linux, although the experience gained there would likely speed things up for this project.

**Comment by LRFLEW** [comment]
Thank you for responding @janisozaur. I'll go ahead and add in some more info.

Yes, Linux support is planned. So is OS X. However, there is a lot that needs to be done before support can be even attempted. You need to keep in mind that, originally, it was thought that Linux support wouldn't be available for OpenRCT2 until after full-decompilation (which hasn't even happened yet). With the wizardry that janisozaur implemented, that hurtle was moved back to where there were no more dependent libraries for the .exe. That milestone took well over a year to reach, and OpenMC2 will not be any different.

In terms of what hurdles are in the way, some of what janisozaur is accurate. The MS ABI won't be too problematic to work around (even if we have to implement it ourselves). DirectX will likely require completely reimplementing the functions that use it so that we can insert an OpenGL interface. There might be some shortcuts we can take with that (DirectX uses a C++ object model that we could, in theory, overload with an OpenGL approximation), but I wouldn't hold your breath. The other **major** hurdle will be phasing out the MSS and Bink libraries, as I cannot get my hands on a linux (or OS X) build of those easily. I have some idea of what I'll do about those, but I need to get further into the project to know what needs to be done for sure.

In short: It's planned, but you will need to use Wine for quite some time.

**Comment by janisozaur** [comment]
Regarding DirectX: I don't claim to have any knowledge on how to use it properly, but there are d3d9 state trackers available/coming which you could possibly use to significantly reduce the initial work required.
- https://wiki.ixit.cz/d3d9
- https://github.com/disks86/SchaeferGL
- wine also has some good d3d9 support

**Comment by Krutonium** [comment]
For now, use Wine. It already actually loads as opposed to the original executable which just crashes immediately with a weird error - https://files.gitter.im/OpenRCT2/OpenRCT2/non-dev/VqAE/screenFetch-2016-06-30_07-32-42.png

![OpenMC2 running on Arch Linux](https://files.gitter.im/OpenRCT2/OpenRCT2/non-dev/VqAE/screenFetch-2016-06-30_07-32-42.png)

**Comment by LRFLEW** [comment]
Wine has been tested with OpenMC2, and will be the preferred method of running it for the time being. Since native support cannot reasonably be achieved currently, and the README has been updated with info about using Wine, I am closing this issue.
